Error when trying to use "Align Region" feature

**Describe the bug**
When opening AutoSplit and using the "Align Region" option, I get an error when selecting a picture (picture attached).

**To Reproduce**
Steps to reproduce the behavior:

1. Open AutoSplit
2. Click on "Align Region"
3. Chose the picture
4. See error

**Expected behavior**
AutoSplit should align the region by comparing the Align Image with the Capture Region, as it did in all versions before v2.0.0.

**Screenshots and recordings**
![image](https://user-images.githubusercontent.com/89664355/234746399-9faf1649-5b3a-416b-ad03-0599b0dcc19b.png)

**Traceback or Crash Report**
If AutoSplit showed an exception traceback, please paste it here:

```
Traceback (most recent call last):

  File "AutoSplit.py", line 181, in <lambda>

  File "region_selection.py", line 194, in align_region

  File "region_selection.py", line 249, in __test_alignment

cv2.error: OpenCV(4.7.0) D:\a\opencv-python\opencv-python\opencv\modules\imgproc\src\templmatch.cpp:1164: error: (-215:Assertion failed) (depth == CV_8U || depth == CV_32F) && type == _templ.type() && _img.dims() <= 2 in function 'cv::matchTemplate'

```
<!-- If AutoSplit crashed and closed without saying anything, please check Windows' Event Viewer, under Windows Logs > Application, for any recent error relating to AutoSplit -->

**Version (please complete the following information):**

- OS: Windows 10.0.19045
- AutoSplit: v2.0.1

**AutoSplit Profile and Split Images**
[toml settings and Align picture.zip](https://github.com/Toufool/AutoSplit/files/11339053/toml.settings.and.Align.picture.zip)

--- a/src/region_selection.py
+++ b/src/region_selection.py
@@ -235,12 +235,10 @@ def __test_alignment(capture: cv2.Mat, template: cv2.Mat):  # pylint: disable=to
     best_width = 0
     best_loc = (0, 0)
 
-    # If the number of color channels don't match, we can run cv2.matchTemplate()
-    if capture.shape[2] != template.shape[2]:
-        if capture.shape[2] == 4:
-            capture = capture[:, :, :3]
-        if template.shape[2] == 4:
-            template = template[:, :, :3]
+    # Add alpha channel to template if it's missing. The cv2.matchTemplate() function
+    # needs both images to have the same color dimensions, and capture has an alpha channel
+    if template.shape[2] == 3:
+        template = cv2.cvtColor(template, cv2.COLOR_BGR2BGRA)
 
     # This tests 50 images scaled from 20% to 300% of the original template size
     for scale in np.linspace(0.2, 3, num=56):

--- a/src/region_selection.py
+++ b/src/region_selection.py
@@ -234,7 +234,14 @@ def __test_alignment(capture: cv2.Mat, template: cv2.Mat):  # pylint: disable=to
     best_height = 0
     best_width = 0
     best_loc = (0, 0)
-
+    
+    # Strip excess color channels if they are present
+    if capture.shape[2] != template.shape[2]:
+        if capture.shape[2] == 4:
+            capture = capture[:, :, :3]
+        if template.shape[2] == 4:
+            template = template[:, :, :3]
+    
     # This tests 50 images scaled from 20% to 300% of the original template size
     for scale in np.linspace(0.2, 3, num=56):
         width = int(template.shape[1] * scale)

--- a/src/region_selection.py
+++ b/src/region_selection.py
@@ -234,14 +234,14 @@ def __test_alignment(capture: cv2.Mat, template: cv2.Mat):  # pylint: disable=to
     best_height = 0
     best_width = 0
     best_loc = (0, 0)
-    
-    # Strip excess color channels if they are present
+
+    # If the number of color channels don't match, we can run cv2.matchTemplate()
     if capture.shape[2] != template.shape[2]:
         if capture.shape[2] == 4:
             capture = capture[:, :, :3]
         if template.shape[2] == 4:
             template = template[:, :, :3]
-    
+
     # This tests 50 images scaled from 20% to 300% of the original template size
     for scale in np.linspace(0.2, 3, num=56):
         width = int(template.shape[1] * scale)

